Sources/Build/Command.compile(ClangModule).swift

```diff
@@ -60,14 +60,14 @@ struct ClangModuleBuildMetadata {
     var inputs: [String] {
         return module.recursiveDependencies.flatMap { module in
             switch module {
-            case let module as ClangModule:
-                 let product = Product(name: module.name, type: .Library(.Dynamic), modules: [module])
-                return product.targetName
-            case let module as CModule:
-                return module.targetName
-            case let module as SwiftModule:
-                return module.targetName
-            default:
+              case let module as ClangModule:
+                let product = Product(name: module.name, type: .Library(.Dynamic), modules: [module])
+                return prefix.appending(product.outname).asString
+              case _ as CModule:
```

Comment:
Oh cool. Still learning about the intricacies of switch. I will look into that and file a bug on the compiler if appropriate.

Oh and I'll fix the non-standard indentation too. Old habits...